controllers/construct/mongodbstatefulset.go

```diff
@@ -128,23 +129,34 @@ func BuildMongoDBReplicaSetStatefulSetModificationFunction(mdb MongoDBStatefulSe
 }
 
 func mongodbAgentContainer(automationConfigSecretName string, volumeMounts []corev1.VolumeMount) container.Modification {
+	agentCommand := strings.Join([]string{
+		"agent/mongodb-agent",
+		"-cluster=" + clusterFilePath,
+		"-skipMongoStart",
+		"-noDaemonize",
+		"-healthCheckFilePath=" + agentHealthStatusFilePathValue,
+		"-serveStatusPort=5000",
+		"-useLocalMongoDbTools"}, " ")
 	return container.Apply(
 		container.WithName(AgentName),
 		container.WithImage(os.Getenv(AgentImageEnv)),
 		container.WithImagePullPolicy(corev1.PullAlways),
 		container.WithReadinessProbe(DefaultReadiness()),
 		container.WithResourceRequirements(resourcerequirements.Defaults()),
 		container.WithVolumeMounts(volumeMounts),
-		container.WithCommand([]string{
-			"agent/mongodb-agent",
-			"-cluster=" + clusterFilePath,
-			"-skipMongoStart",
-			"-noDaemonize",
-			"-healthCheckFilePath=" + agentHealthStatusFilePathValue,
-			"-serveStatusPort=5000",
-			"-useLocalMongoDbTools",
-		},
-		),
+		container.WithSecurityContext(container.DefaultSecurityContext()),
+		container.WithCommand([]string{"/bin/bash", "-c", `current_uid=$(id -u)
+echo $current_uid
+declare -r current_uid
+if ! grep -q "${current_uid}" /etc/passwd ; then
+sed -e "s/^mongodb:/builder:/" /etc/passwd > /tmp/passwd
+echo "mongodb:x:$(id -u):$(id -g):,,,:/:/bin/bash" >> /tmp/passwd
+cat /tmp/passwd
+export NSS_WRAPPER_PASSWD=/tmp/passwd
+export LD_PRELOAD=libnss_wrapper.so
+export NSS_WRAPPER_GROUP=/etc/group
+fi
```

Comment:
This is sadly needed as otherwise the agent will panic with `unknown user 2000`, we will ask the agent team if they can handle it on their side in the future but for now we will have to keep it.

Reply:
should we make a ticket for this now so it doesn't get overlooked?

Reply:
I think so, we need to figure out exactly what we require from the agent team to write a good ticket for them, maybe we can syncup later on zoom and write it together?